python/cudf/cudf/core/frame.py

```diff
@@ -2688,7 +2688,13 @@ def acos(self):
                     1.5707963267948966,  1.266103672779499],
                     dtype='float64')
         """
-        return self._unaryop("acos")
+        if self.iloc[0].dtype == "int":
```

Comment:
Wondering if `is_integer_dtype` is useful here. @shwina ?

Reply:
We're guaranteed that the object on the left hand side is a `dtype`, so it should be safe to compare it with `"int"`. Utilities like `is_integer_dtype` are preferable when dealing with arbitrary objects. For example:

```
np.int32(1) != "int"
# but:
is_integer_dtype(np.int32(1)) == True
```

Reply:
Cool! I'll leave it as is, in that case, if that sounds good to you both :D